src/openfl/_internal/renderer/context3D/Context3DGraphics.hx

```diff
@@ -74,7 +74,7 @@ class Context3DGraphics
 					}
 					else
 					{
-						bitmap = c.shaderBuffer.shader.__bitmap.input;
+						bitmap = c.shaderBuffer.inputs[0];
```

Comment:
It probably is right to use the cached inputs instead of referencing the shader again -- but what if the shader has more than one input? Is it always index 0?

Thanks :smile:

Reply:
From my understanding, we only use `bitmap` in this function to calculate UVs. And what we pass to `drawQuads` (first argument) provides only one set of UVs per quad...
So to support multiple UVs here we first need to make a new `drawQuads`? :thinking:

Reply:
A custom shader can take from multiple shader inputs. It will only have one set of texture UVs, but it still is possible for someone to sample from more than one texture.

For example, say you had a color texture and an alpha texture, and sampled from both to create the final result. There are many more complex things a user can achieve. Most shaders will probably have the one input only, but they may have more. Maybe we can iterate over inputs to find the matching one?

Reply:
Yes, I intended to use multi-texturing too. But I thought OpenFL was using UVs from the first bitmap?

> Maybe we can iterate over inputs to find the matching one?

What is the concept of "matching input"? How OpenFL is intended to determine which input holds the UVs we need? Is there a flag the user passes to mark some input as primary?